src/mt_protocol.cpp

```diff
@@ -254,6 +262,8 @@ void mt_protocol_check_packet(uint32_t now) {
 
   if (pb_buf[0] != MT_MAGIC_0 || pb_buf[1] != MT_MAGIC_1) {
     d("Got bad magic");
+    memset(pb_buf, 0, PB_BUFSIZE);
+    pb_size = 0;
```

Comment:
I'm now thinking it may be safer to only clear one bit at a time, and shift the data over to be sure we don't lose anything important
```suggestion
    if (pb_size > 1) {
        memmove(pb_buf, pb_buf + 1, pb_size - 1);
        pb_size -= 1;
    } else {
        pb_size = 0;
    }
```

Reply:
Maybe it doesn't matter, with some debugging this gives us:
```b_buf (first few bytes): 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 C0 
Got bad magic
pb_buf (first few bytes): 0 0 0 0 0 0 0 0 0 0 0 0 0 0 C0 
Got bad magic
pb_buf (first few bytes): 0 0 0 0 0 0 0 0 0 0 0 0 0 C0 
Got bad magic
pb_buf (first few bytes): 0 0 0 0 0 0 0 0 0 0 0 0 C0 
Got bad magic
pb_buf (first few bytes): 0 0 0 0 0 0 0 0 0 0 0 C0 
Got bad magic
pb_buf (first few bytes): 0 0 0 0 0 0 0 0 0 0 C0 
Got bad magic
pb_buf (first few bytes): 0 0 0 0 0 0 0 0 0 C0 
Got bad magic
pb_buf (first few bytes): 0 0 0 0 0 0 0 0 C0 
Got bad magic
pb_buf (first few bytes): 0 0 0 0 0 0 0 C0 
Got bad magic
pb_buf (first few bytes): 0 0 0 0 0 0 C0 
Got bad magic
pb_buf (first few bytes): 0 0 0 0 0 C0 
Got bad magic
pb_buf (first few bytes): 0 0 0 0 C0 
Got bad magic
pb_buf (first few bytes): 0 0 0 C0 
Got bad magic
pb_buf (first few bytes): 0 0 C0 94 C3 0 2 40 1 
Got bad magic
pb_buf (first few bytes): 0 C0 94 C3 0 2 40 1 
Got bad magic
pb_buf (first few bytes): C0 94 C3 0 2 40 1 
Got bad magic
pb_buf (first few bytes): 94 C3 0 2 40 1 
Got a payloadVariant we don't recognize: 8
pb_buf (first few bytes): 94 C3 0 27 1A 25 8 93 BA 9C 83 F 58 F8 EB 1 
pb_buf (first few bytes): 94 C3 0 27 1A 25 8 93 BA 9C 83 F 58 F8 EB 1 
pb_buf (first few bytes): 94 C3 0 7F 22 7D 8 93 BA 9C 83 F 12 4C A 9 
pb_buf (first few bytes): 94 C3 0 7F 22 7D 8 93 BA 9C 83 F 12 4C A 9
 ```

instead of:

```
pb_buf (first few bytes): 94 C3 0 8B 22 88 1 8 D4 87 D1 D2 D 12 5F A 
pb_buf (first few bytes): 94 C3 0 8B 22 88 1 8 D4 87 D1 D2 D 12 5F A 
pb_buf (first few bytes): 94 C3 0 8B 22 88 1 8 D4 87 D1 D2 D 12 5F A 
Decoding failed
pb_buf (first few bytes): 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
Got bad magic
pb_buf (first few bytes): 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
Got bad magic
pb_buf (first few bytes): 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
Got bad magic
pb_buf (first few bytes): 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
Got bad magic
pb_buf (first few bytes): 94 C3 0 2 40 1 
Got a payloadVariant we don't recognize: 8
pb_buf (first few bytes): 94 C3 0 27 1A 25 8 93 BA 9C 83 F 58 F8 EB 1 
pb_buf (first few bytes): 94 C3 0 27 1A 25 8 93 BA 9C 83 F 58 F8 EB 1 
pb_buf (first few bytes): 94 C3 0 27 1A 25 8 93 BA 9C 83 F 58 F8 EB 1 ```

Reply:
Yeah, I think you can just delete everything, it might also be there was a real serial connection issue.